### PR TITLE
Tutorial room 3

### PR DESCRIPTION
**Pull request by JackPlaymint**
This PR adds a third room to the tutorial.
It demonstrates moving your unit, constructing a building, crafting an item, using quests and potentially adding an item to your player wallet.

The quests are set up to guide the player through the moving and construction tasks and demonstrate a range of quest types.

### CODE REVIEW

**Comment by ldunnplaymint** [comment]
Screenshot 2024-03-19 at 16 07 20

Whenever I run this, I just get an empty map: Also the tiles don't extend far enough so I can spawn and walk to the island.

**Comment by JackPlaymint** [comment]
> Screenshot 2024-03-19 at 16 07 20
> Whenever I run this, I just get an empty map: Also the tiles don't extend far enough so I can spawn and walk to the island.

Do you get a quest? You should just have an empty map with a quest.

I didn't include the centre island as I thought that would be there by default and these spokes would be added after that initial centre island was made.

**Comment by ldunnplaymint** [comment]
Screenshot 2024-03-19 at 16 13 59

My bad, I do get the quest but I am unable to move the Unit without those two extra tiles. Likely worth chucking them in, I think if the HUB yaml has them too, one set just gets dropped anyway.

**Comment by JackPlaymint** [comment]
> I added the tiles locally and I am happy with the room. Once they go in, will tick and merge.

Awesome, just pushed those up now :)

File: contracts/src/maps/tutorial-room-3/building-kinds/CastleOfTreasure.js [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+
+function unitIsBuildingOwner(mobileUnit, selectedBuilding) {
+    //console.log('unit owner id:',  mobileUnit?.owner?.id, 'building owner id:', selectedBuilding?.owner?.id);
+    return mobileUnit?.owner?.id && mobileUnit?.owner?.id === selectedBuilding?.owner?.id;
+}
+
+function unitIsBuildingAuthor(mobileUnit, selectedBuilding) {
+    //console.log('unit owner id:',  mobileUnit?.owner?.id, 'building author id:', selectedBuilding?.kind?.owner?.id);
+    return mobileUnit?.owner?.id && mobileUnit?.owner?.id === selectedBuilding?.kind?.owner?.id;
+}
```
Comment (chrisfarms) [inline contracts/src/maps/tutorial-room-3/building-kinds/CastleOfTreasure.js:133]: think we should remove all the unused code in these plugins to keep the examples concise

**Review by chrisfarms — COMMENTED** [review]
in @Steedie's PR (and related to #1077 ) .. we have said we are going to avoid nesting folding within each map from now on, as it makes working with tile-fab harder and easier to see what is in each example

So could we ensure there are no folders inside tutorial-room-3 (sorry for the faff!)

**Comment by JackPlaymint** [comment]
> in @Steedie's PR (and related to #1077 ) .. we have said we are going to avoid nesting folding within each map from now on, as it makes working with tile-fab harder and easier to see what is in each example
> 
> So could we ensure there are no folders inside tutorial-room-3 (sorry for the faff!)

Oh yeah! No worries

**Pull request merged**